gslib/tests/test_cp.py

```diff
@@ -1060,6 +1062,25 @@ def test_request_reason_header_persists_multiple_requests(self):
                     r' b/this_is_env_reason.*\\r\\n\\r\\n')
     self.assertRegex(stderr, reason_regex)
 
+  @SequentialAndParallelTransfer
+  @SkipForJSON('JSON API uses a different debug log format.')
+  def test_request_reason_header_persists_multiple_requests_xml(self):
+    """Test that x-goog-request-header works when cp sends multiple requests."""
+    os.environ['CLOUDSDK_CORE_REQUEST_REASON'] = 'b/this_is_env_reason'
+    bucket_uri = self.CreateBucket()
+    dst_uri = suri(bucket_uri, 'foo')
+    fpath = self._get_test_file('test.gif')
+
+    boto_config_for_test = ('GSUtil', 'resumable_threshold', '0')
+    with SetBotoConfigForTest([boto_config_for_test]):
+      stderr = self.RunGsUtil(['-D', 'cp', fpath, dst_uri], return_stderr=True)
+
+    reason_regex = (
+        r'Final headers: \{[\s\S]*\''
+        r'x-goog-request-reason\': \'b/this_is_env_reason\'[\s\S]*}')
+
+    self.assertRegex(stderr, reason_regex + r'[\s\S]*' + reason_regex)
```

Comment:
This regex might look a bit odd to someone who doesn't know what the debug output looks like.
Maybe put a one-line comment saying why `reason_regex` comes up twice?